sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -515,7 +492,8 @@ int64_t CurlSession::Read(Azure::Core::Context const& context, uint8_t* buffer,
   // Also if we have already read all contentLength
   if (this->m_sessionTotalRead == this->m_contentLength || this->m_rawResponseEOF)
   {
-    // Read everything already
+    // make sure EOF for response is set to true
+    this->m_rawResponseEOF = true;
```

Comment:
If you forcabley kill the connection in the dtor, then maybe you can get rid of this field